modules/iam-primary-roles/versions.tf

```
@@ -1,18 +1,10 @@
 terraform {
-  required_version = ">= 0.12.0"
+  required_version = ">= 1.0.0"
 
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 2.0"
-    }
-    template = {
-      source  = "hashicorp/template"
-      version = ">= 2.0"
-    }
-    local = {
-      source  = "hashicorp/local"
-      version = ">= 1.3"
+      version = ">= 3.0"
```

This should be consistent with the others

```suggestion
      version = "~> 4"
```